client/concurrentlocalstore.go

```diff
@@ -0,0 +1,49 @@
+package client
+
+import (
+	"encoding/json"
+	"sync"
+)
+
+// ConcurrentLocalStore provides a LocalStore over an existing LocalStore
+// implementation that is safe for concurrent access.
+type ConcurrentLocalStore struct {
```

Comment:
No particular reason except it's flexible. Any `LocalStore` can be protected with this construct, and the existing ones did not perform any concurrency control, so I figured it can be solved for them all with this. And It _can_ simplify development as the locking is managed by a different layer. I'm fine to refactor if it's desired.

There can definitely be some scenarios that does not require concurrency control, but I hardly think there is a measurable performance benefit from omitting it :)

Reply:
Yeah, intuitively, I would think that if no concurrency control is actually needed, there would be zero lock contention, and thus minimal overhead for the default safety.